text/0004-new-pre-coordicide-consensus/0004-new-pre-coordicide-consensus.tex.md

```diff
@@ -0,0 +1,191 @@
+   - If it is valid but conflicting with the current state then place it in an ordered *Conflict Set*.
+4. **Local overdraft handling**: After you are done traversing the subtangle attempt to apply again the conflicting bundles. Iterate the *Conflict Set* in order and attempt to apply the bundles to the ledger state. Mark the applied bundles as `seen` and `approved`. Remove applied bundles from the set. Repeat this step until the *Conflict Set* size doesn't change and mark the remaining bundles of the set as seen and ignored.
+
+**Note**
+Once a bundle is marked as ignored/seen/approved this will be final and it can't be changed by a later milestone that comes in.
+
+
+## Node Tip Selection
+
+Due to white-flag, no part of the tangle can be censored, thus the tangle can't be split by a double-spend. So as long as each new tip (bundle) a user creates approves two other random tips, the tangle shouldn't get divided into several subtangles. Thus we can get have a very fast tip selection by just selecting random tips.
+
+The problem is that lazy users may approve bundles or transactions that are not tips... This is a problem becuase lazy behavior can hinder confirmation rates. A lazy tip approves less bundles in the tangles. So we should have defenses in place to make sure that such lazy transactions will be left behind by honest tip-selection.
+
+So instead of total random tip selection we will do a weighted random tip selection. We will give each tip a score:
+
+```
+0 - If it is a lazy tip that shouldn't be selected
+1 - If it is a somewhat lazy tip
+2 - If it is a non-lazy good tip.
+```
+Since the actual consensus will be determined by milestone tip selection there can be more than one way to distribute the scores.
+We will propose two similar score systems that can be used by node implementations, a timestamp based one, and a milestone based one.
+
+But first some definitions:
+
+`Tip`:
+A solid bundle tail of a bundle that has no approvers.
+
+`Approved transaction Roots`:
+All `seened` (by milestone) transaction that can be reached by walking from a given transaction down to its parents. The walk must terminate once we reached a `seened` transaction.
+
+`Transaction Snapshot Index`:
+The index of the milestone that marked the transaction as `seen`
+
+`Oldest Transaction Root Snapshot`:
+The milestone bundle with the lowest index that marked any of the Approved Transaction Roots as `seen`.
+
+`Youngest Transaction Root Snapshot`:
+The milestone bundle with the highest index that marked any of the Approved Transaction Roots as `seen`.
+
+
+
+### Timestamp based scoring 
+
+#### Configurable Values
+$C_1$ - Time in ms that a tip's timestamp can be *below* its solidification time.
+
+$C'_1$ - Time in ms that a tip's timestamp can be *above* its soldification time.
+
+$C_2$ - Max difference between tip solidification time and parent bundle solidification timestamp.
```

Comment:
For clarity I think this should say "approvee" instead of "parent bundle"

Reply:
I thought we are now busy destroying CFB's legacy and this "approvee" made up word is part of it :trollface: 
Anyhows, since I see you are also one of the "approvee" fans I'll change it